core/src/main/java/org/elasticsearch/rest/support/RestUtils.java

```diff
@@ -238,4 +242,45 @@ public static Pattern checkCorsSettingForRegex(String corsSetting) {
 
         return null;
     }
+
+    /**
+     * URL encode an HTTP response header value.  Non-ascii
```

Comment:
@jpountz the issue with URLEncoder was that it would URL encode everything other than a-zA-Z0-9 and a few other characters ('_', '-', '*', maybe one other).  This was causing headers like content-type: application/json to come back as application%2Fjson.  I don't know if browsers can handle that and in the very least, it was causing some of our tests to fail.